src/nsresourced/nsresourcework.c

```diff
@@ -1760,6 +1760,14 @@ static int vl_method_add_netif_to_user_namespace(sd_varlink *link, sd_json_varia
                 return -ENOMEM;
         strshorten(ifname_host, IFNAMSIZ-1);
 
+        r = userns_info_add_netif(userns_info, ifname_host);
```

Comment:
Should this be done before creating netif?

If the host interface already exists, then that causes the conflicting interface will be owned this userns.

I think it is better to add the entry after the interface is created.

Reply:
So, the host-side ifname is hashed from unique data, so it should really not happen that we collide there with anything by accident. (it's 64bit, not 128bit, admittedly, but still. we do it within the `ns-*` ifname 'namespace', so we'd clash only against ourselves here)

One of the reasons I went for the hashed names (the other was that ifnames are so short) is precisely the chance of reducing collisions. 

The reason I prefer adding this to the db first is mostly that we definitely clean up the ifnames if anything later fails. i.e. i generally think it's a good thing to ensure destruction of some object is going to work, before actually allocating it.

Hence, I really think we should leave it like this..

(Will add comment about this)